snuba/query/snql/__init__.py

```diff
@@ -0,0 +1,36 @@
+from snuba.datasets.dataset import Dataset
+from snuba.query.logical import Query
+from parsimonious.grammar import Grammar
+
+
+grammar = Grammar(
+    r"""
+    query_exp = "MATCH" overall_clause
+
+    overall_clause = match_clause where_clause? collect_clause? having_clause? order_by_clause?
+
+    match_clause = space* "(" clause ")" space*
+    where_clause = space* "WHERE" clause space*
+    collect_clause = space* "COLLECT" clause "BY" clause space*
+    having_clause = space* "HAVING" clause space*
+    order_by_clause = space* "ORDER BY" clause ("ASC"/"DESC") space*
+
+    clause = space* ~r"[-=><\w]+" space*
+    space = " "
+"""
+)
+
+exp_tree = grammar.parse(
+    "MATCH (cake) WHERE icing=chocolate COLLECT soon BY thursday HAVING candles ORDER BY names ASC"
+)
+print(exp_tree)
```

Comment:
please remove this from here.
Feel free to add it to a test when you add the visitor. I doubt it would be viable or useful to test the parse tree alone.